packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/EditInvoiceDetails.tsx

```diff
@@ -0,0 +1,287 @@
+          : null,
+        purchase_order: invoice.purchase_order ?? '',
+
+        /** Items section */
+        line_items: invoice.line_items.map((lineItem) => ({
+          quantity: lineItem.quantity,
+          product_id: lineItem.product.id,
+          vat_rate_id: lineItem.product.vat_rate.id,
+          vat_rate_value: lineItem.product.vat_rate.value,
+          name: lineItem.product.name,
+          price: lineItem.product.price,
+          measure_unit_id: lineItem.product.measure_unit_id,
+        })),
+        vat_exemption_rationale: invoice.vat_exemption_rationale ?? '',
+
+        /** Items section */
+        entity_bank_account_id: invoice.entity_bank_account?.id ?? '',
+        payment_terms_id: invoice.payment_terms?.id ?? '',
+      }),
+      [
+        counterpartShippingAddress?.id,
+        invoice.counterpart_contact,
+        invoice.counterpart_id,
+        invoice.counterpart_vat_id?.id,
+        invoice.entity_bank_account?.id,
+        invoice.entity_vat_id?.id,
+        invoice.fulfillment_date,
+        invoice.line_items,
+        invoice.payment_terms?.id,
+        invoice.purchase_order,
+        invoice.vat_exemption_rationale,
+      ]
+    ),
+  });
+
+  const [actualCurrency, setActualCurrency] = useState(invoice.currency);
+
+  const {
+    handleSubmit,
+    setValue,
+    formState: { isDirty },
+  } = methods;
+
+  useEffect(() => {
+    setValue('default_shipping_address_id', counterpartShippingAddress?.id);
+  }, [counterpartShippingAddress?.id, setValue]);
+
+  const [isAlertOpen, setIsAlertOpen] = useState(false);
+  const handleCancelWithAlert = useCallback(() => {
+    /** If the form hasn't been changed, we could just cancel it */
+    if (!isDirty) {
+      onCancel();
+
+      return;
+    }
+
+    /** Otherwise open cancel alert modal */
+    setIsAlertOpen(true);
+  }, [isDirty, onCancel]);
+
+  const updateReceivableLineItems = useUpdateReceivableLineItems(invoice.id);
+  const updateReceivable = useUpdateReceivable(invoice.id);
+
+  const isLoading =
+    updateReceivableLineItems.isPending || updateReceivable.isPending;
+
+  return (
+    <MoniteStyleProvider>
+      <DialogTitle>
+        <Toolbar>
+          <Button
+            variant="text"
+            color={isDirty ? 'error' : 'primary'}
+            onClick={handleCancelWithAlert}
+            startIcon={<ArrowBackIcon />}
+            disabled={isLoading}
+          >{t(i18n)`Cancel`}</Button>
+          <Box sx={{ marginLeft: 'auto' }}>
+            <Button
+              variant="contained"
+              key="next"
+              color="primary"
+              type="submit"
+              form="receivablesDetailsForm"
+              disabled={isLoading}
+            >{t(i18n)`Update`}</Button>
+          </Box>
+        </Toolbar>
+      </DialogTitle>
+      <Divider />
+      <DialogContent>
+        <FormProvider {...methods}>
+          <form
+            id="receivablesDetailsForm"
```

Comment:
It would be better to replace `receivablesDetailsForm` with the unique ID:
```ts
const formId = `Monite-Form-${useId()}`
```

Probably, you can add similar changes to [EditInvoiceDetails](https://github.com/team-monite/monite-sdk/blob/9ceb783e8a09c1b63355e25675b34fade03f16fe/packages/sdk-react/src/components/receivables/InvoiceDetails/ExistingInvoiceDetails/components/EditInvoiceDetails.tsx#L182)

Reply:
Hm, but why? We always use specific form ids

Reply:
There may be a situation the `id` on the customer's page will conflict with the non-unique `id` of `receivablesDetailsForm`

Reply:
Let's add this topic to FE sync. Because it's a general change and if it really possible to have any problem with it we could change it but globally for all forms then

Reply:
It seems better to avoid obvious mistakes in advance than to correct them in the future.

Reply:
@MadMed677, By the way, why do we need to use `id`?

Reply:
@nekedos When we submit the form we have to provide which form we want to submit. 
Example: https://github.com/team-monite/monite-sdk/blob/61ee304aa3066640b0a49c78d735cbdf0ac4d1b5/packages/sdk-react/src/components/approvalPolicies/ApprovalPolicyDetails/ApprovalPolicyDetailsForm/ApprovalPolicyDetailsForm.tsx#L274

Reply:
@radist2s updated